utils/doclint/generateDotnetApi.js

```diff
@@ -0,0 +1,695 @@
+    const requiredSuffix = argument.required ? "" : isEnum ? " = default" : " = null";
+    args.push(`${innerArgType}${requiredPrefix} ${innerArgName}${requiredSuffix}`);
+  };
+
+  let parseArg = (/** @type {Documentation.Member} */ arg) => {
+    if (arg.name === "options") {
+      arg.type.properties.forEach(prop => {
+        parseArg(prop);
+      });
+      return;
+    }
+
+    if (arg.type.expression === '[string]|[path]') {
+      let argName = translateMemberName('argument', arg.name, null);
+      pushArg("string", argName, arg);
+      pushArg("string", `${argName}Path`, arg);
+      if (arg.spec) {
+        addParamsDoc(argName, XmlDoc.renderTextOnly(arg.spec, maxDocumentationColumnWidth));
+        addParamsDoc(`${argName}Path`, [`Instead of specifying <paramref name="${argName}"/>, gives the file name to load from.`]);
+      }
+      return;
+    } else if (arg.type.expression === '[boolean]|[Array]<[string]>') {
+      // HACK: this hurts my brain too
+      // we split this into two args, one boolean, with the logical name
+      let argName = translateMemberName('argument', arg.name, null);
+      let leftArgType = translateType(arg.type.union[0], parent, (t) => { throw 'Not supported'; });
+      let rightArgType = translateType(arg.type.union[1], parent, (t) => { throw 'Not supported'; });
+
+      pushArg(leftArgType, argName, arg);
+      pushArg(rightArgType, `${argName}Values`, arg);
+
+      addParamsDoc(argName, XmlDoc.renderTextOnly(arg.spec, maxDocumentationColumnWidth));
+      addParamsDoc(`${argName}Values`, [`The values to take into account when <paramref name="${argName}"/> is <code>true</code>.`]);
+
+      return;
+    }
+
+    const argName = translateMemberName('argument', arg.alias || arg.name, null);
+    const argType = translateType(arg.type, parent, (t) => generateNameDefault(member, argName, t, parent));
+
+    if (argType === null && arg.type.union) {
+      // we might have to split this into multiple arguments
+      let translatedArguments = arg.type.union.map(t => translateType(t, parent, (x) => generateNameDefault(member, argName, x, parent)));
+      if (translatedArguments.includes(null))
+        throw 'Unexpected null in translated argument types. Aborting.';
+
+      let argDocumentation = XmlDoc.renderTextOnly(arg.spec, maxDocumentationColumnWidth);
+      for (const newArg of translatedArguments) {
+        const sanitizedArgName = newArg.match(/(?<=^[\s"']*)(\w+)/g, '')[0] || newArg;
+        const newArgName = `${argName}${sanitizedArgName[0].toUpperCase() + sanitizedArgName.substring(1)}`;
+        pushArg(newArg, newArgName, arg);
+        addParamsDoc(newArgName, argDocumentation);
+      }
+      return;
+    }
+
+    addParamsDoc(argName, XmlDoc.renderTextOnly(arg.spec, maxDocumentationColumnWidth));
+
+    if (argName === 'timeout' && argType === 'decimal') {
+      args.push(`int timeout = 0`); // a special argument, we ignore our convention
```

Comment:
timeout is double everywhere else, should be decimal in c# too I believe.

Reply:
why is a double, can we pass milliseconds with decimals?